derive/src/derive.rs

```diff
@@ -53,7 +53,7 @@ fn variant_attrs(attrs: &[syn::Attribute]) -> Attrs {
         })
 }
 
-pub fn tovalue_derive(mut s: synstructure::Structure) -> proc_macro::TokenStream {
+pub fn intovalue_derive(mut s: synstructure::Structure) -> proc_macro::TokenStream {
```

Comment:
@zshipko question, why did you go from `to` -> `into`? (I'm curious about this because I went the opposite way in ocaml-interop, by making `into_rust` -> `to_rust` because I think that in rust the `into` assumes that `self` is consumed.

Reply:
Ah this is something I probably want to change back. In this case the trait is actually consuming `self`, but I don't think that is really the best idea here.

Reply:
Yes, the thing is that `Value` is `Copy` right? I think I had it like that for the same reason, but since the original value is still available I decided to switch to the `to_` prefix. I am not really sure about what the convention is in cases like this, but my reasoning is that (in addition to what I said on the previous sentence) since `Value` is more like a reference than the actual value `to_` made more sense.

Reply:
Now I remember that `IntoValue` consumes `self` to make `Custom` values better integrated. It seems like consuming `self` is actually kind of nice because it prevents the value from being accessed after being converted to OCaml.

I will need to think about the various trade-offs a little more...

Reply:
> Yes, the thing is that `Value` is `Copy` right? I think I had it like that for the same reason, but since the original value is still available I decided to switch to the `to_` prefix. I am not really sure about what the convention is in cases like this, but my reasoning is that (in addition to what I said on the previous sentence) since `Value` is more like a reference than the actual value `to_` made more sense.

The main impetus for the name change was that clippy was complaining about it being named `To*` and not taking a reference. But I think your explanation makes more sense.